Common/Systems/MobSystem/MobMappingSystem.cs

```diff
@@ -1,164 +0,0 @@
-//		_npcs.Remove(npc);
-//	}
-
-//	public override void PreUpdateNPCs()
-//	{
-//		if (!SubworldSystem.AnyActive())
-//		{
-//			foreach (NPC npc in _npcs)
-//			{
-//				DespawnNpc(npc);
-//			}
-
-//			return;
-//		}
-
-//		List<NPC> npcsToDespawn = [];
-//		foreach (NPC npc in _npcs)
-//		{
-//			foreach (Player player in Main.ActivePlayers)
-//			{
-//				if (player.Center.Distance(npc.Center) > 1800f) // TODO: spawn distance hard coded; make it smartly, somehow...
-//				{
-//					npcsToDespawn.Add(npc);
-//					break;
-//				}
-//			}
-//		}
-
-//		List<NPC> npcsToSpawn = [];
-//		foreach (KeyValuePair<NPC, Vector2> npc in _savedNpcs)
-//		{
-//			foreach (Player player in Main.ActivePlayers)
-//			{
-//				if (player.Center.Distance(npc.Value) < 1200f) // TODO: spawn distance hard coded; make it smartly, somehow...
-//				{
-//					npcsToSpawn.Add(npc.Key);
-//					break;
-//				}
-//			}
-//		}
-
-//		npcsToDespawn.ForEach(DespawnNpc);
-//		npcsToSpawn.ForEach(SpawnNpc);
-//	}
-
-//	// copied from NPC class
-//	private static int GetAvailableNPCSlot(int Type, int startIndex)
-//	{
-//		bool num = Type >= 0 && NPCID.Sets.SpawnFromLastEmptySlot[Type];
-//		int t = 200;
-//		int num2 = 1;
-//		if (num)
-//		{
-//			t--;
-//			Utils.Swap(ref startIndex, ref t);
-//			num2 = -1;
-//		}
-
-//		for (int i = startIndex; i != t; i += num2)
-//		{
-//			if (!Main.npc[i].active)
-//			{
-//				return i;
-//			}
-//		}
-
-//		for (int j = startIndex; j != t; j += num2)
-//		{
-//			if (Main.npc[j].CanBeReplacedByOtherNPCs)
-//			{
-//				return j;
-//			}
-//		}
-
-//		return -1;
-//	}
-
-//	private static readonly MethodInfo _method = typeof(NPCLoader).GetMethod("OnSpawn", BindingFlags.NonPublic | BindingFlags.Static);
-//	public static void MakeNPC(Vector2 center, int type, IEntitySource source = null)
-//	{
-//		NPC npc = new MapNPC();
-//		npc.SetDefaults(type);
-//		// npc.whoAmI = availableNPCSlot; || no slot for now
-//		// GiveTownUniqueDataToNPCsThatNeedIt(type, availableNPCSlot); || we wont be spawning town npcs, i think
-//		npc.Center = center;
-//		npc.active = true;
-//		// npc.timeLeft = int.MaxValue; || it wont despawn to inactivity
-//		npc.wet = Collision.WetCollision(npc.position, npc.width, npc.height);
-//		npc.ai[0] = 0f;
-//		npc.ai[1] = 0f;
-//		npc.ai[2] = 0f;
-//		npc.ai[3] = 0f;
-//		npc.target = 255;
-
-//		/*
-//		if (type == 50)
-//		{
-//			if (Main.netMode == 0)
-//			{
-//				Main.NewText(Language.GetTextValue("Announcement.HasAwoken", npc.typeName), 175, 75);
-//			}
-//			else if (Main.netMode == 2)
-//			{
-//				ChatHelper.BroadcastChatMessage(NetworkText.FromKey("Announcement.HasAwoken", npc.GettypeNetName()), new Color(175, 75, 255));
-//			}
-//		}
-//		*/
-
-//		// NPCLoader.OnSpawn(npc, source);
-//		_method.Invoke(null, [npc, source]);
-
-//		_savedNpcs.Add(npc, npc.Center);
-//	}
-//}
-
-//internal class MapNPC : NPC
-//{
-//	public new bool DoesntDespawnToInactivity()
-//	{
-//		return true;
-//	}
-//}
```

Comment:
Should this really be removed? I feel like this could be helpful when we inevitably creating the full mapping system

Reply:
The code here is written so poorly that it'd be entirely rewritten anyway. None of this file makes any sense in the tMod API, and doesn't make much sense out of it either.